kong.conf.default

```diff
@@ -447,6 +447,16 @@
                                  # See the `lua_ssl_trusted_certificate`
                                  # setting to specify a certificate authority.
 
+#pg_semaphore_max = 0            # Sets the number of semaphore resources made
+                                 # available to the semaphore used to limit
+                                 # concurrent connections to the Postgres
+                                 # database. The default value of 0 removes the
+                                 # semaphore concurrency limiter.
```

Comment:
Should we rename this property to something more explicit for end-users? E.g. `pg_max_concurrent_queries` or so. Ultimately, "semaphore" is an implementation detail, and "max semaphore resources" does not provide a hint on what said resources are used for.

Also, we should note in the comment that this property will limit concurrent queries/connections on a per-worker basis only.

Reply:
I had conflicting thoughts, and the details are spelled out in the initial commit message. The config here is indeed part of an implementation detail, but that's a detail that users need to be aware of. `pg_max_concurrent_queries` as a config option, to me, would read as the number of concurrent queries that a Kong node could issue, which is not accurate. Something more accurate like `pg_max_worker_concurrent_queries` seems like a mouthful, and is a bit clunky- it almost reads like it's talking about a pg worker, which isn't really a thing.

So, I'd vote we keep the names relevant to the implementation detail to avoid confusion, and spell out well (with examples, diagrams, etc) the impacts of various configs in the documentation.

Reply:
My thinking here was to use `pg_max_concurrent_queries` and specify in the comment for this property that this is a per-worker value. There is precedent here already with the `lua_socket_pool_size` property (which is, alas, not specified as being "per-worker" which I believe is an oversight).

Reply:
I've updated the variable name per this suggestion. To me it makes the following `pg_semaphore_timeout` name a bit incongruous as it also doesn't hide an implementation detail? But maybe it's worth the detail here? :man_shrugging: